src/service/document-click.js

```diff
@@ -180,6 +195,43 @@ export class ClickHandler {
   }
 
   /**
+   * Handle out going link decoration
+   * @param {!Event} e
+   * @param {!Element} target
+   * @param {!Location} tgtLoc
+   * @private
+   */
+  handleExternalNavDecoration_(e, target, tgtLoc) {
+    // Check outgoing link within whitelist domain
+    if (!this.whiteListOutgoingDomain_) {
+      this.whiteListOutgoingDomain_ = {};
+      const cUrl = Services.documentInfoForDoc(this.ampdoc).canonicalUrl;
```

Comment:
Should you whitelist the AMP URL in addition to the canonical? Their domains can be different.